src/components/App.jsx

```diff
@@ -11,80 +11,117 @@ class App extends React.Component {
 		};
 	}
 
-	changeText = (text) => {
-		this.setState({
-			text: text,
-		});
-	};
-
 	toogleProductToCart = (id) => {
-		this.state.products.map((item) => {
+		const { cart, products } = this.state;
+		products.map((item) => {
+			// I need this "item"...
 			if (item.id === id) {
-				if (this.state.cart.cartProducts.includes(item)) {
+				if (!!cart.cartProducts.find((prod) => prod.id === id)) {
 					this.setState({
 						cart: {
-							// удаляем товар из корзины
-							cartProducts: this.state.cart.cartProducts.filter((prod) => {
-								return prod.id !== id;
-							}),
-							// уменьшаем тотал
-							totalPrice: this.state.cart.totalPrice - item.price * item.amount,
+							cartProducts: cart.cartProducts.filter((prod) => prod.id !== id),
+							totalPrice: cart.totalPrice - item.price * item.amount,
 						},
+						products: products.map((prod) => {
+							if (prod.id === id) {
+								prod.left = prod.left + prod.amount;
+								prod.amount = 0;
+							}
+							return prod;
+						}),
 					});
 				} else {
 					this.setState({
 						cart: {
-							// добавляем товар в корзину
-							cartProducts: [...this.state.cart.cartProducts, item],
-							// добавялем прайс к тоталу
-							totalPrice: this.state.cart.totalPrice + item.price * item.amount,
+							// ...here, that's why I didn't use destructurization
+							cartProducts: [...cart.cartProducts, item],
+							totalPrice: cart.totalPrice + item.price,
 						},
+						products: products.map((prod) => {
+							if (prod.id === id) {
+								prod.amount = prod.amount + 1;
+								prod.left = prod.left - 1;
+							}
+							return prod;
+						}),
 					});
 				}
 			}
 		});
-		console.log(this.state.cart.cartProducts);
 	};
 
 	increment = (id) => {
+		const { cart, products } = this.state;
 		this.setState({
 			cart: {
-				// долбавляем количесвто продукта в корзине
-				cartProducts: this.state.cart.cartProducts.map((prod) => {
+				cartProducts: cart.cartProducts.map((prod) => {
 					if (prod.id === id) {
 						prod.amount = prod.amount + 1;
 					}
 					return prod;
 				}),
-				// totalPrice: this.state.cart.totalPrice + prod.price,
+				totalPrice: cart.totalPrice + cart.cartProducts.find((prod) => prod.id === id).price,
+			},
+			products: products.map((prod) => {
+				if (prod.id === id) {
+					prod.left = prod.left - 1;
+				}
+				return prod;
+			}),
+		});
+	};
+
+	decrement = (id) => {
+		const { cart, products } = this.state;
+		this.setState({
+			cart: {
+				cartProducts: cart.cartProducts.map((prod) => {
+					if (prod.id === id) {
+						prod.amount = prod.amount - 1;
+					}
+					return prod;
+				}),
+				totalPrice: cart.totalPrice - cart.cartProducts.find((prod) => prod.id === id).price,
 			},
+			products: products.map((prod) => {
+				if (prod.id === id) {
+					prod.left = prod.left + 1;
+				}
+				return prod;
+			}),
 		});
 	};
 
 	render() {
+		const { text, cart, products } = this.state;
+
+		const props = {
```

Comment:
you have a lot of props, please think about the logic, the feeling that it's all left with the parent